### ISSUE TITLE
Force update not working

### ISSUE THREAD

**Issue by tryallthethings** [ravibpatel/AutoUpdater.NET#678]
Hello! I've encountered an issue while trying to implement a force update button. I started the app, said "Remind me later" for the update popup and then clicked the button for the forced update.

```C#
private void btnCheckUpdate_Click(object sender, EventArgs e)
{
    Log.Info("Started manual search for updates");
    AutoUpdater.Mandatory = true;
    AutoUpdater.UpdateMode = Mode.ForcedDownload;
    AutoUpdater.Start("https://www.domain.com/downloads/update.xml");
}
```

But this doesn't do anything. It logs the message but doesn't execute the update. I had to add another parameter that wasn't in the [#README.md ](https://github.com/ravibpatel/AutoUpdater.NET/blob/c5b8cfb0508c1efcd130f3859fba10cd6a6789eb/README.md?plain=1#L192) at the forced update section. With this, the download immediately starts and runs the update.

```C#
private void btnCheckUpdate_Click(object sender, EventArgs e)
{
    Log.Info("Started manual search for updates");
    AutoUpdater.Mandatory = true;
    AutoUpdater.UpdateMode = Mode.ForcedDownload;
    AutoUpdater.ShowRemindLaterButton = false;
    AutoUpdater.Start("https://www.domain.com/downloads/update.xml");
}
```

Is this a bug or by design (and the README.md isn't up-to-date)?

And is there a way to get any log messages from Autoupdater.NET that I can feed into Nlog for debugging purposes?

**Comment by ravibpatel** [comment]
Both code you provided should work same. As you can see [here](https://github.com/ravibpatel/AutoUpdater.NET/blob/master/AutoUpdater.NET/AutoUpdater.cs#L447). It skips Remind Later check and turns that button visibility off. Also, ForcedDownload mode will trigger download directly upon update and don't ask user for any confirmation. In your case, hitting the button should trigger the download.

Currently, there is no log created by AutoUpdater.NET. Only ZipExtractor creates log.

**Comment by tryallthethings** [comment]
The first code block didn't do anything. No download, no popups, no errors. As soon as I add in `AutoUpdater.ShowRemindLaterButton = false;`, the download immediately starts. I tried it multiple times at the time of creating the issue and could reproduce it this way reliably.
I tried it again today. And of course I'm not able to reproduce it that way. I love Software development. Today it works even without the extra option just fine. I'll get back to you if I'm able to reproduce it again
